frontend/packages/console-dynamic-plugin-sdk/src/extensions/details-item.ts

```diff
@@ -0,0 +1,63 @@
+import { ExtensionK8sModel } from '../api/common-types';
+import { Extension, ExtensionDeclaration, CodeRef } from '../types';
+import { K8sResourceCommon } from './console-types';
+
+export type DetailsItemColumn = 'right' | 'left';
+
+export type DetailsItemComponentProps<R extends K8sResourceCommon = K8sResourceCommon> = {
+  /** The resource for which details are being rendered. */
+  obj: R;
+};
```

Comment:
Can we also pass the `path?: string;` and `value?: any/never` to the component?

1. The path from the extensions would allow us to have one "DateTime" component that could be used for two fields, for example, a "Start time" and "Completion time", etc.
2. Wdyt about extracting and passing also the `value` from `obj`? When rendering the Component you use already `_.get(obj, path, '-')` to extract it when there is no Component. That would allow us to make a reused component even easier. In the example from above, this means Just consuming the DateTime string from the props.

```suggestion
  /** The resource for which details are being rendered. */
  obj: R;

  /**
   * The fully qualified path to a string-type property on the resource as defined in the extension.
   * @example 'spec.template.spec.containers[0].image'
   */
  path?: string;

  /**
   * If the path is defined, the value contains the extracted value from the Kubernetes resource.
   */
  value?: never;
```